.github/workflows/mypy.yaml

```diff
@@ -43,6 +43,7 @@ jobs:
         run: |
           python -m pip install --no-deps git+https://github.com/OpenFreeEnergy/gufe@main
           python -m pip install mypy
+          python -m pip install types-pkg_resources
```

Comment:
This is a new one for me, what does this do?

Reply:
It installs typing stubs for pkg_resources, without it mypy throws an error. It got introduced in this PR since I use a utility from it to compare package versions

Reply:
see this error message https://github.com/OpenFreeEnergy/openfe/actions/runs/9028999373/job/24810539868